lib/mocha_runner.coffee

```diff
@@ -0,0 +1,19 @@
+ArgumentBuilder = require './argument_builder'
+mochaExecutablePath = require.resolve('mocha/bin/mocha')
+{spawn} = require 'child_process'
+
+
+class MochaRunner
+
+  constructor: ({argv, cwd}) ->
+    @argumentBuilder = new ArgumentBuilder {argv, cwd}
+
+
+  run: (done) ->
+    @argumentBuilder.build (err, args) ->
+      if err then return done err
+      childProcess = spawn mochaExecutablePath, args, stdio: 'inherit'
```

Comment:
We would have to do all the argument parsing ourselves then. 
Mocha doesn't have a API where you can pass in the argv, thats all done in the file located in bin.